### PR TITLE
Create unit test for exam progress endpoint

### LINKED ISSUE #3546: create unit test for exam progress endpoint
### Observed behavior

Described here: https://github.com/learningequality/kolibri/pull/3499#issuecomment-379827352

Let's create a test to make sure the exam progress always works for various input values

### Expected behavior

Slightly complex logic has test coverage

### User-facing consequences

NA

### PR DESCRIPTION
**Pull request by mrpau-richard**
### Summary
I created a unit test for the exam progress endpoint. 
### References
This fixes #3546 
### Contributor Checklist

- [x] Contributor has fully tested the PR manually
- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label

### Reviewer Checklist

- [ ] Automated test coverage is satisfactory
- [ ] Reviewer has fully tested the PR manually
- [ ] PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- [ ] External dependencies files were updated (`yarn` and `pip`)
- [ ] Documentation is updated
- [ ] Link to diff of internal dependency change is included
- [ ] CHANGELOG.rst is updated for high-level changes
- [ ] Contributor is in AUTHORS.rst

### CODE REVIEW

File: kolibri/logger/test/test_api.py [hunk 1]
```diff
@@ -387,6 +387,11 @@ def setUp(self):
         self.user2 = FacilityUserFactory.create(facility=self.facility)
         self.exam = Exam.objects.create(title="", channel_id="", question_count=0, collection=self.facility, creator=self.user2, active=True)
         self.examlog = ExamLog.objects.create(exam=self.exam, user=self.user1)
+        [ExamAttemptLog.objects.create(
```
Comment (jonboiser) [inline kolibri/logger/test/test_api.py]: Could you format this differently? One argument per line:

```
ExamAttemptLog.objects.create(
    item="d4623921a2ef5ddaa39048c0f7a6fe06",
    examlog=self.examlog,
    ...
)
```

File: kolibri/logger/test/test_api.py [hunk 2]
```diff
@@ -387,6 +387,11 @@ def setUp(self):
...
+            item="d4623921a2ef5ddaa39048c0f7a6fe06", examlog=self.examlog, user=self.user1,
+            content_id=uuid.uuid4().hex, channel_id=uuid.uuid4().hex, start_timestamp=str(datetime.datetime.now()),
+            end_timestamp=str(datetime.datetime.now()), correct=0
```
Comment (jonboiser) [inline kolibri/logger/test/test_api.py]: To make this somewhat more realistic, could you "jitter" these timestamps a bit? Maybe you could use the index in the `for x in in range(3)` to make these timestamps slightly different.

File: kolibri/logger/test/test_api.py [hunk 3]
```diff
@@ -387,6 +387,11 @@ def setUp(self):
         self.user2 = FacilityUserFactory.create(facility=self.facility)
         self.exam = Exam.objects.create(title="", channel_id="", question_count=0, collection=self.facility, creator=self.user2, active=True)
```
Comment (jonboiser) [inline kolibri/logger/test/test_api.py]: If you can, could you improve this test by making `question_count = 1`, then adjusting your `ExamAttempLogs` accordingly?

**Comment by jonboiser** [comment]
Could you squash all of your commits into one?

**Comment by rtibbles** [comment]
Squashed and merged.

**Pull request merged**